File: daily_challenges/717.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function(bits) {
    const n = bits.length
    let i = 0
    while(i < n - 1){
        i += (bits[i] === 1) ? 2 : 1
    }
    return i === n - 1
};


/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function (bits) {
    let n = bits.length
    let count1 = 0
    for (let i = n - 2; i >= 0 && bits[i] === 1; i--) {
        count1++
    }
    return count1 % 2 === 0
};

File: daily_challenges/717.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        i = 0
        while i < n - 1:
            i += 2 if bits[i] == 1 else 1
        return i == n - 1

class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        count1 = 0
        i = n - 2
        while i >= 0 and bits[i] == 1:
            count1 += 1
            i -= 1
        return count1 % 2 == 0
